Junit/src/IPL.java

```diff
@@ -0,0 +1,364 @@
+        List<Player> playerSearchList = listOfPlayers.stream().filter(p -> p.getName().equals(query) || p.getName().split(" ")[0].equals(query) || p.getName().split(" ")[1].equals(query)).collect(Collectors.toList());
+
+        for (Player player : playerSearchList) {
+            System.out.println(player.getName());
+        }
+
+    }
+
+    // find highest wicket taken or runs scored in a given team
+    public static void findHighest(List<Player> listOfPlayers,String team){
+        int higestWicketByPlayer = 0;
+        int highestRunScoredByPerson = 0;
+        int maxNameSize = 20;
+        char[] maxWicketPlayerName = new char[maxNameSize];
+        char[] maxRunScorerName = new char[maxNameSize];
+
+        for (Player player : listOfPlayers) {
+            if(player.getTeam().equals(team)){
+
+                if(higestWicketByPlayer < player.getWicket()){
+                    higestWicketByPlayer = player.getWicket();
+                    maxWicketPlayerName = player.getName().toCharArray();
+                }
+
+                if(highestRunScoredByPerson < player.getRuns()){
+                    highestRunScoredByPerson = player.getRuns();
+                    maxRunScorerName = player.getName().toCharArray();
+                }
+            }
+        }
+        System.out.println("Highest wicket taker is " +  String.valueOf(maxWicketPlayerName));
+        System.out.print("Highest run scorer is " + String.valueOf(maxRunScorerName));
+    }
+
+    // finding top three batsmen bowler and all rounder
+    // all rounder is found using a cumulative score of runs and wickets.
+    public static void findTopThree(List<Player> listOfPlayers){
+
+        List<Player> battingList = listOfPlayers.stream().filter(p -> p.getRole().equals("BATSMAN")).collect(Collectors.toList());
+        Collections.sort(battingList,new RunComparator());
+
+        System.out.println("===================== Batsman ================");
+
+        for(int i=0;i<3;i++){
+            System.out.print(battingList.get(i).getName() + " " + battingList.get(i).getRuns() + "\n");
+        }
+
+        System.out.println("===================== Bowler ================");
+
+        List<Player> bowlingList = listOfPlayers.stream().filter(p -> p.getRole().equals("BOWLER")).collect(Collectors.toList());
+        Collections.sort(bowlingList,new WicketComparator());
+
+        for(int i=0;i<3;i++){
+            System.out.print(bowlingList.get(i).getName() + " " + bowlingList.get(i).getWicket() + "\n");
+        }
+
+        System.out.println("============= All Rounders ==================");
+
+
+        List<Player> allRounderList = listOfPlayers.stream().filter(p -> p.getRole().equals("ALL ROUNDER")).collect(Collectors.toList());
+        Collections.sort(allRounderList,new RunWicketComparator());
+
+        for(int i=0;i<3;i++){
+            System.out.println(allRounderList.get(i).getName() + " " + allRounderList.get(i).cumulativeScore());
```

Comment:
could have used streams themselves to limit the number of results to 3.
Now it is open to IndexOutOfBoundException since you do not really know if the result list has 3 elements in it